### ISSUE TITLE
bcftools merge segfault V1.9

### ISSUE THREAD

**Issue by curtisjamison** [samtools/bcftools#1084]
We're getting a segfault when attempting to merge three files. One VCF has blocks, the other two are annotated files. The segfault occurs very early in the process. Any suggestions of help on this matter would be greatly appreciated.

bcftools 1.9
htslib 1.9

The command we are using is:
/home/bcftools/bin/bcftools merge --force-samples -m both -g /mnt/data/refData/GRCh38/GRCh38.fa 556022.vcf.gz 556024.vcf.gz 576348.vcf.gz

The output of the command is:
````
1       708852  .       T       TA      .       LowGQX;LowDepth BLOCKAVG_min30p3a       GT:GQX:DP:DPF:MIN_DP:RC:AC:GP:DS        0/0:3:1:0:1:.:.:.:.     0/0:.:.:.:.:0:0:1,1e-10,1e-10:3e-10     0/0:.:.:.:.:0:0:1,1e-10,1e-10:3e-10
1       708853  .       a       .       .       LowGQX;LowDepth END=708903;BLOCKAVG_min30p3a    GT:GQX:DP:DPF:MIN_DP    0/0:3:1:0:1     ./.:.:.:.:.     ./.:.:.:.:.
1       708904  .       G       .       .       LowGQX  END=708925;BLOCKAVG_min30p3a    GT:GQX:DP:DPF:MIN_DP    .:.:0:0:0       .:.:.:.:.       .:.:.:.:.
1       708926  .       A       .       .       LowGQX;LowDepth END=709075;BLOCKAVG_min30p3a    GT:GQX:DP:DPF:MIN_DP    0/0:3:1:0
Segmentation fault (core dumped)
````

Lookin in the region where the segfault occurred, we have the following lines in the files:
````
556022.vcf.gz:
1       708904  .       G       .       .       LowGQX  END=708925;BLOCKAVG_min30p3a    GT:GQX:DP:DPF:MIN_DP    .:.:0:0:0
1       708926  .       A       .       .       LowGQX;LowDepth END=709075;BLOCKAVG_min30p3a    GT:GQX:DP:DPF:MIN_DP    0/0:3:1:0:1
1       709076  .       A       .       .       LowGQX  END=722049;BLOCKAVG_min30p3a    GT:GQX:DP:DPF:MIN_DP    .:.:0:0:0
1       722050  .       G       .       .       LowGQX;LowDepth END=722074;BLOCKAVG_min30p3a    GT:GQX:DP:DPF:MIN_DP    0/0:3:1:0:
--
556024.vcf.gz:
1       708143  .       C       CA      .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       708852  .       T       TA      .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       710045  .       A       G       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       710225  rs185127847     T       A       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:0.979012,0.0208983,8.93246e-05:0.0210769
1       710330  ALU_umary_ALU_2 A       <INS:ME:ALU>    .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       711151  rs371268074     CTAT    C       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       713463  .       TATTA   T       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       713812  rs201942322     A       T       .       LOWCONF .       GT:RC:AC:GP:DS  0/0:0:0:0.808777,0.187484,0.00373891:0.194962
1       715779  .       C       T       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       715852  .       C       T       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       717447  .       G       GA      .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       722312  .       C       T       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
--
576348.vcf.gz:
1       708143  .       C       CA      .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       708852  .       T       TA      .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       710045  .       A       G       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       710225  rs185127847     T       A       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:0.979012,0.0208983,8.93246e-05:0.0210769
1       710330  ALU_umary_ALU_2 A       <INS:ME:ALU>    .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       711151  rs371268074     CTAT    C       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       713463  .       TATTA   T       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       713812  rs201942322     A       T       .       LOWCONF .       GT:RC:AC:GP:DS  0/0:0:0:0.808777,0.187484,0.00373891:0.194962
1       715779  .       C       T       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       715852  .       C       T       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       717447  .       G       GA      .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
1       722312  .       C       T       .       PASS    .       GT:RC:AC:GP:DS  0/0:0:0:1,1e-10,1e-10:3e-10
--
````

**Comment by pd3** [comment]
Thank you for the bug report. It would be helpful if you could provide a test case with full headers.

**Comment by curtisjamison** [comment]
Hi Petr -

I have sent an email with links to the files that create the segfault directly to you.

Thanks,
Curt

**Comment by pd3** [comment]
As discussed offline, this seems to be fixed by c33f8827974.
